drivers/serial/serial.c

```diff
@@ -1370,24 +1326,69 @@ static ssize_t uart_readv(FAR struct file *filep, FAR struct uio *uio)
 #endif
 
   nxmutex_unlock(&dev->recv.lock);
-  if (recvd >= 0)
+  return recvd;
+}
+
+static ssize_t uart_read(FAR struct file *filep,
+                         FAR char *buffer, size_t buflen)
+{
+  return uart_read_internal(filep, buffer, buflen, false);
+}
+
+/****************************************************************************
+ * Name: uart_readv
+ ****************************************************************************/
+
+static ssize_t uart_readv(FAR struct file *filep, FAR struct uio *uio)
+{
+  FAR const struct iovec *iov = uio->uio_iov;
+  int iovcnt = uio->uio_iovcnt;
+  bool nonblock = false;
```

Comment:
need check flags in filep

Reply:
which flags?

Reply:
O_NONBLOCK

Reply:
uart_read will check nonblock together with f_oflags